File: DesignPatternsExercises/ruby/mediator_design_pattern/exercise/mediator_design_pattern.rb
require_relative 'mediator'
require_relative 'seller'
require_relative 'buyer'

mediator = Mediator.new
Seller.new(:eurSeller, mediator, :USD, 100)
Seller.new(:gbpSeller1, mediator, :GBP, 50)
Seller.new(:gbpSeller2, mediator, :GBP, 100)
Seller.new(:inrSeller1, mediator, :INR, 100)
Seller.new(:inrSeller2, mediator, :INR, 200)

gbp_buyer = Buyer.new(mediator, :GBP)
gbp_buyer.attempt_to_purchase(79)
puts ''
gbp_buyer.attempt_to_purchase(79)
puts ''
inr_buyer = Buyer.new(mediator, :INR)
inr_buyer.attempt_to_purchase(20_000)
puts ''
rus_buyer = Buyer.new(mediator, :RUS)
rus_buyer.attempt_to_purchase(1)

File: DesignPatternsExercises/java/MediatorDesignPattern/Exercise/MediatorDesignPattern.java
public class MediatorDesignPattern {
	public static void main(String[] args) {
		Mediator mediator = new Mediator();
		new Seller("eurSeller", mediator, "EUR", 100);
		new Seller("gbpSeller1", mediator, "GBP", 50);
		new Seller("gbpSeller2", mediator, "GBP", 100);
		new Seller("inrSeller1", mediator, "INR", 100);
		new Seller("inrSeller2", mediator, "INR", 200);

		Buyer gbpBuyer = new Buyer(mediator, "GBP");
		gbpBuyer.attemptToPurchase(79);
		System.out.println();
		gbpBuyer.attemptToPurchase(79);
		System.out.println();
		Buyer inrBuyer = new Buyer(mediator, "INR");
		inrBuyer.attemptToPurchase(20000);
		System.out.println();
		Buyer rusBuyer = new Buyer(mediator, "RUS");
		rusBuyer.attemptToPurchase(1);
	}
}
